tests/terminate/pos/Div.hs

```diff
@@ -0,0 +1,19 @@
+module Div where
+
+{-@ iterateTo0 :: n:Int -> {v:Int | v == 0} @-}
+iterateTo0 :: Int -> Int
+iterateTo0 n
+  | n <= 0    = 0
+  | otherwise = iterateTo0 (div n 2)
+
+{-@ test :: {v:Int | v == 0} @-}
+test = iterateTo0 (-1)
```

Comment:
How about adding a test that also checks that non-termination is correctly detected? Something like

```haskell
{-@ fail iterateTo0 @-}
iterateTo0 :: Int -> Int
iterateTo0 0 = 0
iterateTo0 n = iterateTo0 (div n 2)
```

or it could be a neg test

```haskell
{-@ LIQUID "--expect-error-containing=..." @-}

{-@ fail iterateTo0 @-}
iterateTo0 :: Int -> Int
iterateTo0 0 = 0
iterateTo0 n = iterateTo0 (div n 2)
```

Reply:
Added a test for this in last commit. Thanks!